packages/web3-providers-http/src/index.js

```diff
@@ -97,7 +97,7 @@ HttpProvider.prototype.send = function (payload, callback) {
     var request = this._prepareRequest();
 
     request.onreadystatechange = function() {
-        if (request.readyState === 4 && request.timeout !== 1) {
+        if (request.readyState === 4 && request.timeout !== 1 && (request.status >= 200 && request.status < 300)) {
```

Comment:
@uluhonolulu Do you think this change is necessary? I'm worried this could break someone code

Do you think we could change it to (as `request.status` is `0` mentioned [here](https://github.com/ChainSafe/web3.js/issues/3425)):
```suggestion
        if (request.readyState === 4 && request.timeout !== 1 && request.status !== 0) {
```

Reply:
Good idea, I'll do that.

Reply:
@spacesailor24 Did you mean that you would do it yourself? I tried to push my change but it doesn't show up here.

Reply:
@uluhonolulu Yea, sorry about that, since I merged your PR into my branch you won't be able to push changes - just wanted your feedback